### PR TITLE
Hao - Creating a warning window for double checking when deleting a commons

### LINKED ISSUE #7: FEATURE: Add caution before deleting Commons
# User Story

* As an admin
* I can see a "are you sure" popup when deleting a commons
* So that I don't accidentally delete an entire game in progress

### Discussion

Currently, the Delete button on the commons page is a little dangerous.

If you click it, an entire commons goes away, with no confirmation dialog.

It would be better to add a modal (i.e. a popup) that asks the user: Are you sure you want to delete this commons?

The modal would have options such as [Permanently Delete] and [Keep this Commons].

### Acceptance Criteria

- [x] If you click Delete on a commons, you get a modal, and the commons is not immediately deleted.
- [x] When you get the modal, if you click the [Keep this Commons] button, you are returned to the commons list page, and nothing is deleted
- [x] When you get the modal, if you click the [Delete this Commons] button, the commons is deleted.

### Implementation Todos

An explanation of React Bootstrap modals can be found here: https://react-bootstrap.netlify.app/docs/components/modal/

### PR DESCRIPTION
**Pull request by haowu0521**
In this PR, I created a warning window that shows "Cancel" and "Delete" for doubling checking when Admin tries to delete a commons.
image
Closes #7

### CODE REVIEW

**Review by EthanSun210 — DISMISSED** [review]
LGTM

File: frontend/src/main/components/Commons/CommonsTable.js [hunk 1]
```diff
@@ -92,9 +104,35 @@ export default function CommonsTable({ commons, currentUser }) {
 
     const columnsToDisplay = hasRole(currentUser,"ROLE_ADMIN") ? columnsIfAdmin : columns;
 
-    return <OurTable
-        data={commons}
-        columns={columnsToDisplay}
-        testid={testid}
-    />;
+    // return <OurTable
```
Comment (GDW1) [inline frontend/src/main/components/Commons/CommonsTable.js]: commented out code, cant merge with it in PR

File: frontend/src/tests/components/Commons/CommonsForm.test.js [hunk 2]
```diff
@@ -85,7 +86,38 @@ describe("CommonsForm tests", () => {
     expect(screen.getByTestId("CommonsForm-Submit-Button")).toHaveTextContent("Create New Commons");
 
 
+    //fireEvent.click(submitButton);
```
Comment (GDW1) [inline frontend/src/tests/components/Commons/CommonsForm.test.js]: Commented out code

File: frontend/src/tests/components/Commons/CommonsForm.test.js [hunk 3]
```diff
@@ -94,26 +126,21 @@ describe("CommonsForm tests", () => {
...
+    //   "CommonsForm-startingDate",
+    //   "CommonsForm-degradationRate",
+    //   "CommonsForm-carryingCapacity",
+    // ].forEach(
+    //   (testid) => {
+    //     const element = screen.getByTestId(testid);
+    //     expect(element).toBeInTheDocument();
+    //     expect(element).toHaveClass("is-invalid");
+    //   }
+    // );
```
Comment (GDW1) [inline frontend/src/tests/components/Commons/CommonsForm.test.js]: commented out code

**Review by GDW1 — CHANGES_REQUESTED** [review]
Change the commented out code. Everything else looks fine

File: frontend/src/tests/components/Commons/CommonsForm.test.js (hunk 3 above)
Comment (haowu0521) [reply frontend/src/tests/components/Commons/CommonsForm.test.js]: deleted commented out code

**Review by GDW1 — DISMISSED** [review]

**Pull request merged**